fvm/environment/programs.go

```diff
@@ -288,9 +295,18 @@ type dependencyStack struct {
 }
 
 func newDependencyStack() *dependencyStack {
-	return &dependencyStack{
+	stack := &dependencyStack{
 		trackers: make([]dependencyTracker, 0),
 	}
+
+	// The root of the stack is the program (script/transaction) that is being executed.
+	// At the end of the transaction execution, this will hold all the dependencies
+	// of the script/transaction.
+	//
+	// The root of the stack should never be popped.
+	stack.push(common.StringLocation("ProgramDependencyStackRoot"))
```

Comment:
can you add some invalid/weird symbols characters to the string, just in case?  e.g., "^ProgramDependencyStackRoot$"